### DISCUSSION TITLE
How would you do add ExpandAllRows and CollapseAllRows to a Table?

### DISCUSSION THREAD

**Discussion by jpgarza93** [ant-design-blazor/ant-design-blazor#2286] [Q&A]
1. Is there a command I can call to expand or collapse all rows?

2. Is there a way for me to get all the Rows from the Table object once it's created?

I was only able to achieve collapse/expand all this way:

```
@{ CustomerTableRows = new(); }

<AntDesign.Table OnRowClick="@((row) => {row.Expanded = !row.Expanded;})">
    <AntDesign.ActionColumn >
        <CellRender Context="cellData">
            @{ CustomerTableRows.Add(cellData.RowData); }
       </CellRender>
</AntDesign.Table>
  
@code{
    List<RowData> CustomerTableRows { get; set; }
    
    private void ExpandAll(bool expand)
    {
        CustomerTableRows.ForEach(r => r.Expanded = expand);
        StateHasChanged();
    }
}
```

But it has its cons, it does not work great with nested tables.

**Comment by daramire** [comment]
![header-expand-all-icon](https://user-images.githubusercontent.com/7254799/156933976-e055c582-f835-4660-a67e-4dff28bc0b9b.png)

Hello, any suggestion on how can I add a expandAll button in header?
